server/channels/app/content_flagging.go

```diff
@@ -710,3 +718,49 @@ func (a *App) publishContentFlaggingReportUpdateEvent(targetId, teamId string, p
 
 	return nil
 }
+
+func (a *App) SaveContentFlaggingConfig(config model.ContentFlaggingSettingsRequest) *model.AppError {
+	err := a.Srv().Store().ContentFlagging().SaveReviewerSettings(config.ReviewerSettings.ReviewerIDsSettings)
+	if err != nil {
+		return model.NewAppError("SaveContentFlaggingConfig", "app.content_flagging.save_reviewer_settings.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+	}
+
+	systemConfig := a.Config().Clone()
+	systemConfig.ContentFlaggingSettings = model.ContentFlaggingSettings{}
+	systemConfig.ContentFlaggingSettings.EnableContentFlagging = config.EnableContentFlagging
+	systemConfig.ContentFlaggingSettings.NotificationSettings = config.NotificationSettings
+	systemConfig.ContentFlaggingSettings.AdditionalSettings = config.AdditionalSettings
+	systemConfig.ContentFlaggingSettings.ReviewerSettings = &model.ReviewerSettings{
+		CommonReviewers:         config.ReviewerSettings.CommonReviewers,
+		SystemAdminsAsReviewers: config.ReviewerSettings.SystemAdminsAsReviewers,
+		TeamAdminsAsReviewers:   config.ReviewerSettings.TeamAdminsAsReviewers,
+	}
+
+	_, _, appErr := a.SaveConfig(systemConfig, true)
+	if appErr != nil {
+		return appErr
+	}
```

Comment:
There is actually a canonical way of updating the config, you can use something like this:
```go
app.App.UpdateConfig(func(cfg *model.Config) {
		cfg.ContentFlaggingSettings.EnableContentFlagging = config.EnableContentFlagging
		cfg.ContentFlaggingSettings.NotificationSettings = config.NotificationSettings
		cfg.ContentFlaggingSettings.AdditionalSettings = config.AdditionalSettings
		cfg.ContentFlaggingSettings.ReviewerSettings = &model.ReviewerSettings{
			CommonReviewers:         config.ReviewerSettings.CommonReviewers,
			SystemAdminsAsReviewers: config.ReviewerSettings.SystemAdminsAsReviewers,
			TeamAdminsAsReviewers:   config.ReviewerSettings.TeamAdminsAsReviewers,
	}
})
```

No need to clone because it can cause a data race issue on other configurations.